pkg/virt-handler/vm.go

```diff
@@ -526,8 +526,12 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 					ifc := v1.VirtualMachineInstanceNetworkInterface{
 						Name: network.Name,
 						IP:   podIface.PodIP,
-						IPs:  []string{podIface.PodIP},
+						IPs:  []string{},
```

Comment:
What is the reason for not using `IPs:  []string{podIface.PodIP}`? Why for loop is needed?

Reply:
its a slice, and if we use hardcoded []string{podIface.PodIPs[0],podIface.PodIPs[1]}
it will add an empty field in the slice, in case of single stuck
(podIface.PodIPs[1] will be empty and will be added to the slice)

Reply:
EDIT: I had a typo, meant - `IPs: []string{podIface.PodIPs}`
Wouldn't it work?

Reply:
nope 2 reasons
1. [] vs [2] 
2. we dont want to add "" if exists, i will fix that by adding if != ""

Reply:
fixed 2 , 1 which is what you asked wont work

Reply:
1. ok, my mistake again - `IPs: podIface.PodIPs`
2. You're creating the array, don't put there empty string in the first place.

Reply:
1. i tried podIface.PodIPs as well (as you mentioned on the one that start with "edit" one), and whatever would be written even if split to elements as in my first answer, it wont work because of 2 
2. already fixed, its a json, it can either say the size or always have two places.
just when reading i check if not nil.

Reply:
I tried it locally and it works. Just don't specify the size [here](https://github.com/kubevirt/kubevirt/pull/3958/files#diff-5e37ab6e114c3110d75e482145f60f8fR49). And `make` the slice with the correct size.

Reply:
please explain exactly because i didnt understand,
when i didnt use size in the podcache it didnt work for me (first part of what u said), but lets try what you said as a whole,
how you suggest to encode it to the json , lts say you know the length of it dynamically in the encoding, according len
(it will change a bit the building but lets see first what you suggest please)